relayer/relayer.go

```diff
@@ -0,0 +1,157 @@
+package relayer
+
+import (
+	"fmt"
+	"math/big"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+	hcf "github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/abi"
+	"github.com/umbracle/ethgo/jsonrpc"
+)
+
+var commitEvent = abi.MustNewEvent(`event NewBundleCommit(uint256 startId, uint256 endId, bytes32 root)`)
+
+type Relayer struct {
+	dataDir           string
+	rpcEndpoint       string
+	stateReceiverAddr ethgo.Address
+	logger            hcf.Logger
+	client            *jsonrpc.Client
+	txRelayer         txrelayer.TxRelayer
+	key               ethgo.Key
+}
+
+func sanitizeRPCEndpoint(rpcEndpoint string) string {
+	if rpcEndpoint == "" || strings.Contains(rpcEndpoint, "0.0.0.0") {
+		_, port, err := net.SplitHostPort(rpcEndpoint)
+		if err == nil {
+			rpcEndpoint = fmt.Sprintf("http://%s:%s", "localhost", port)
+		} else {
+			rpcEndpoint = "http://localhost:8545"
+		}
+	}
+
+	return rpcEndpoint
+}
+
+func NewRelayer(
+	dataDir string,
+	rpcEndpoint string,
+	stateReceiverAddr ethgo.Address,
+	logger hcf.Logger,
+	key ethgo.Key,
+) *Relayer {
+	endpoint := sanitizeRPCEndpoint(rpcEndpoint)
+
+	// create the JSON RPC client
+	client, err := jsonrpc.NewClient(endpoint)
+	if err != nil {
+		logger.Error("Failed to create the JSON RPC client", "err", err)
+
+		return nil
+	}
+
+	txRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithClient(client))
+
+	return &Relayer{
+		dataDir:           dataDir,
+		rpcEndpoint:       endpoint,
+		stateReceiverAddr: stateReceiverAddr,
+		logger:            logger,
+		client:            client,
+		txRelayer:         txRelayer,
+		key:               key,
+	}
+}
+
+func (r *Relayer) Start() error {
+	et := eventTracker{
+		dataDir:           r.dataDir,
+		rpcEndpoint:       r.rpcEndpoint,
+		stateReceiverAddr: r.stateReceiverAddr,
+		subscriber:        r,
+		logger:            r.logger,
+	}
+
+	return et.start()
+}
+
+func (r *Relayer) AddLog(log *ethgo.Log) {
+	r.logger.Info("Received a log", "log", log)
+
+	if commitEvent.Match(log) {
+		vals, err := commitEvent.ParseLog(log)
+		if err != nil {
+			panic(err)
+		}
+
+		var startID uint64
+		if sid, ok := vals["startId"].(big.Int); ok {
+			startID = sid.Uint64()
+		}
+
+		var endID uint64
+		if eid, ok := vals["endId"].(big.Int); ok {
+			endID = eid.Uint64()
+		}
+
+		r.logger.Info("Commit: Block %d StartID %d EndID %d\n", log.BlockNumber, startID, endID)
```

Comment:
Be mindful here, logger is expecting odd number of arguments in order to format it properly. So either use `fmt.Sprintf` and specify single formatted string or apply these changes:

```suggestion
		r.logger.Info("Commit", "Block", log.BlockNumber, "StartID", startID, "EndID", endID)
```